stl/CMakeLists.txt

```diff
@@ -507,6 +507,9 @@ function(add_stl_dlls D_SUFFIX REL_OR_DBG)
 
     add_library(msvcp${D_SUFFIX} SHARED)
     target_link_libraries(msvcp${D_SUFFIX} PRIVATE msvcp${D_SUFFIX}_eha_objects msvcp${D_SUFFIX}_objects msvcp${D_SUFFIX}_init_objects "${TOOLSET_LIB}/vcruntime${D_SUFFIX}.lib" "${TOOLSET_LIB}/msvcrt${D_SUFFIX}.lib" "ucrt${D_SUFFIX}.lib")
+    if(VCLIBS_TARGET_ARCHITECTURE MATCHES "^arm64(ec)?$")

```

Comment:
I repeat in this PR the question if we really need `arm64`. 
This is mostly question to @StephanTLavavej.
I think we could drop non-EC from MSBuild project instead.

Reply:
I blindly copied what VCRuntime was doing. I'll see if this is necessary.

Reply:
I just checked by modifying all occurrences to be ARM64EC-only. The "ARM64 and ARM64EC" pattern is necessary for VCRuntime which otherwise breaks with:

```
setjmpex.obj : error LNK2001: unresolved external symbol _mm_getcsr (EC Symbol)
D:\msvc\binaries\arm64chk\bin\arm64\vcruntime140.dll : fatal error LNK1120: 1 unresolved externals
```

I don't really understand what's going on here or if the STL's usage is different, but since there's no real cost to linking an unnecessary library, I'd prefer to follow the same incantation.